lstmcpipe/utils.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+import subprocess
+
+
+def rerun_cmd(cmd, outfile, max_ntry=2, subdir_failures='failed_outputs', **run_kwargs):
+    """
+    rerun r0_to_dl1 process given by `cmd` as long as the exit code is 0 and number of try < max_ntry
+    move the failed output file to subdir failed_outputs
+
+    Parameters
+    ----------
+    cmd: str
+    simtel_file: Path
+    subdir_failures: str
+    outfile: Path
+        path to the cmd output file
+    max_ntry: int
+    run_kwargs: kwargs for subprocess.run
+    """
+    outfile = Path(outfile)
+    ret = -1
+    ntry = 1
+    while ret != 0 and ntry <= max_ntry:
```

Comment:
ret will never be == 0, since ret is the `CompletedProcess` instance. You either need to check `ret.returncode` or assign `ret.returncode` to ret.

E.g. like this:

```
ret = CompletedProcess(args=[], returncode=-1)

while ret.returncode !=0 or ntry <= max_ntry:
    ...
```

at the momemnt I think you are always executing the script max_ntry even if it succeeds.

Reply:
Well spotted, thanks. Fixed and unit tested with lstchain as well.